### PR TITLE
kubectl: add less verbose version

### PR DESCRIPTION
**Pull request by philips**
The kubectl version output is very complex and makes it hard for users
and vendors to give actionable information. For example during the
recent Kubernetes 1.4.3 TLS security scramble I had to write a one-liner
for users to get out the version number to give to figure out if they
are vulnerable:

```
$ kubectl version | grep -i Server | sed  -n 's%.*GitVersion:"\([^"]*\).*%\1%p'
```

Instead this patch outputs simply output by default

```
./kubectl version
Client Version: v1.4.3
Server Version: v1.4.3
```

Adding the `--verbose` flag will output the old format.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35543)

### CODE REVIEW

**Comment by brendandburns** [comment]
@philips I hate to say this but I think that this is a breaking API change.  It would be fine to add a `--short` flag to get the concise version, but I think there are highly likely scripts out there that are parsing the output of `kubectl version` and will break.

@bgrant0607 @kubernetes/sig-cli for thoughts.

**Comment by philips** [comment]
@brendandburns Yea... how do you version a command line CLI.

**Comment by bgrant0607** [comment]
cc @kubernetes/kubectl @kubernetes/sig-cluster-lifecycle

**Comment by smarterclayton** [comment]
Very, very carefully :)  Basically no breaking changes except on major boundaries.  We do have `-q` in a few places.

**Comment by smarterclayton** [comment]
Although I don't think classic `-q` meets what you describe - `--short` is backwards compatible and probably more correct.

**Comment by bgrant0607** [comment]
see also #20304

**Comment by AdoHe** [comment]
```
 It would be fine to add a --short flag to get the concise version
```
SGTM. I have wrote many scripts to use the old version output, I don't hope to break this.

**Comment by philips** [comment]
re #20304 @bgrant0607 In etcdctl we just started using an environment variable (ETCDCTL_API=X) because providing a flag for output version might be version specific and cause errors...

**Comment by philips** [comment]
Pushed a new version that uses `--short`.

**Comment by fabianofranz** [comment]
+1 to `--short` and not breaking backwards behavior.

File: examples/kubectl-container/Makefile [hunk 1]
```diff
@@ -27,7 +27,7 @@ kubectl:
 	cp ../../_output/local/bin/$(GOOS)/$(GOARCH)/kubectl .
 
 .tag: kubectl
-	./kubectl version --client | grep -o 'GitVersion:"[^"]*"' | sed 's/[^"]*"\([^"+]*\).*/\1/' > .tag
+	./kubectl version --verbose --client | grep -o 'GitVersion:"[^"]*"' | sed 's/[^"]*"\([^"+]*\).*/\1/' > .tag
```
Comment (fabianofranz) [inline examples/kubectl-container/Makefile]: This looks like leftovers from your original proposal. ;)

**Comment by mbruzek** [comment]
@chuckbutler You will be interested in this change as we are parsing the kubectl output in some of our code. Please take a look.

**Comment by bgrant0607** [comment]
If we're implicitly establishing a new output convention here, please document it to make it explicit:

https://github.com/kubernetes/kubernetes/blob/master/docs/devel/kubectl-conventions.md

**Comment by philips** [comment]
@bgrant0607 added convention doc for `--short` https://github.com/kubernetes/kubernetes/pull/35543/commits/1dc64f3c6c21b2d830b8ef9bae72b1583adcea14
@AdoHe @mbruzek default output shouldn't change now.
@smarterclayton use environment variables, flags for version are version problems :) `ETCDCTL_API=3`

**Comment by brendandburns** [comment]
@philips I'd be ok with adding a magic ${QUIET_KUBECTL_VERSION} or somesuch environment variable to opt in to the new behavior...

current code LGTM.

**Comment by brendandburns** [comment]
you need to sync to head and run `./hack/update-bazel.sh`

**Comment by philips** [comment]
@brendandburns ran bazel stuff and squashed!

**Comment by brendandburns** [comment]
@k8s-bot node e2e test this please issue: #35983

**Pull request merged**